code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -1005,7 +1005,7 @@
 /obj/vehicle/sealed/mecha/container_resist_act(mob/living/user)
 	if(isAI(user))
 		var/mob/living/silicon/ai/AI = user
-		if(!AI.can_shunt)
+		if(AI.can_shunt)
```

Comment:
This is not the right fix. `can_shunt` should not be set to false in the first place unless the ai entered through domination.